notificationservice.h

```diff
@@ -145,7 +147,7 @@ class NOTIFYSERVICESHARED_EXPORT NotificationService: public QObject
     QHash<int, Listner> _listners;
     NotificationData _question;
     NotificationData _notify;
-    QList<NotificationData> _history;
+    QSharedPointer<HistoryNotificationsModel> _history;
```

Comment:
I think for this case, better use raw pointers, 
create on constructor and delete on destructor. 
```suggestion
    HistoryNotificationsModel* _history = nullptr;
```